### PR TITLE
LIMS-4350 : Orders: Create: Orders Choose test plans without test units

### PR DESCRIPTION
**Pull request by omnia353**
```
c:\users\omnia\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named 'fcntl'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-03 13:25:34.013 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-03 13:25:34.016 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-03 13:25:34.695 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : bhICccqMaB-AFcn24m2ic_MHHqHVH6ot58vViITYulc .....

DevTools listening on ws://127.0.0.1:61324/devtools/browser/22865dd7-7506-48f0-843f-f43ad3793dea
Orders: Create: Orders Choose test plans without test units ...
2020-09-03 13:26:06.344 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test091_select_large_number_of_test_units_in_one_testplan
2020-09-03 13:26:13.058 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-03 13:26:13.102 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:26:13.855 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:26:14.630 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-03 13:26:15.118 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-03 13:26:15.127 | INFO     | api_testing.apis.orders_api:set_configuration:412 - set order configuration
2020-09-03 13:26:15.597 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-03 13:26:15.604 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_select_large_number_of_test_units_in_one_testplan:2915 - create multiple testplan with same article of each suborder
2020-09-03 13:26:15.616 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:16.002 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.011 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/articles
2020-09-03 13:26:16.164 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.168 | INFO     | api_testing.apis.article_api:get_formatted_article_with_formatted_material_type:207 - search for article with material type {'id': 1, 'text': 'Raw Material'}
2020-09-03 13:26:16.171 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:16.279 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.291 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:16.439 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.448 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/449
2020-09-03 13:26:16.568 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.577 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:16.738 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.747 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:16.856 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:16.861 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:16.991 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.010 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/450
2020-09-03 13:26:17.118 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.134 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:17.272 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.281 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:17.393 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.401 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/articles
2020-09-03 13:26:17.525 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.529 | INFO     | api_testing.apis.article_api:get_formatted_article_with_formatted_material_type:207 - search for article with material type {'id': 1, 'text': 'Raw Material'}
2020-09-03 13:26:17.532 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:17.623 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.632 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:17.779 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.788 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/451
2020-09-03 13:26:17.900 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:17.908 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:18.048 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.051 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:18.139 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.148 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:18.285 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.293 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/452
2020-09-03 13:26:18.405 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.414 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:18.584 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.593 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:18.694 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.702 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/articles
2020-09-03 13:26:18.821 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.827 | INFO     | api_testing.apis.article_api:get_formatted_article_with_formatted_material_type:207 - search for article with material type {'id': 1, 'text': 'Raw Material'}
2020-09-03 13:26:18.830 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:18.923 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:18.940 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:19.090 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.096 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/453
2020-09-03 13:26:19.225 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.234 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:19.520 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.539 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-03 13:26:19.665 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.674 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-03 13:26:19.837 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.854 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/454
2020-09-03 13:26:19.959 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:19.976 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-03 13:26:20.121 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:26:20.130 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_select_large_number_of_test_units_in_one_testplan:2922 -  Create new order.
2020-09-03 13:26:20.139 | INFO     | ui_testing.pages.order_page:create_new_order:124 -  Create new order.
2020-09-03 13:26:20.148 | INFO     | ui_testing.pages.orders_page:click_create_order_button:23 - Press create order button
2020-09-03 13:26:21.261 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:26:30.427 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:26:32.158 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:26:33.883 | INFO     | ui_testing.pages.order_page:set_new_order:20 - Set new order.
2020-09-03 13:26:41.652 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:26:49.752 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:26:56.988 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:27:06.404 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:27:19.952 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:27:37.308 | INFO     | ui_testing.pages.order_page:create_new_order:155 -  Order created with no : 151-2020 
2020-09-03 13:27:37.352 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_select_large_number_of_test_units_in_one_testplan:2927 - create 3 suborder
2020-09-03 13:27:37.404 | INFO     | ui_testing.pages.order_page:create_new_suborder:358 - Add new suborder.
2020-09-03 13:27:38.177 | INFO     | ui_testing.pages.order_page:create_new_suborder:364 - Set material type : Raw Material
2020-09-03 13:27:45.860 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:27:47.346 | INFO     | ui_testing.pages.order_page:create_new_suborder:367 - Set article name : f7a7c378b9
2020-09-03 13:27:47.351 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:27:58.072 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:27:59.032 | INFO     | ui_testing.pages.order_page:create_new_suborder:375 - Set test plan : 1a3ad4f272
2020-09-03 13:28:06.710 | INFO     | ui_testing.pages.order_page:create_new_suborder:375 - Set test plan : f52fc816b9
2020-09-03 13:28:14.161 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:15.093 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:16.197 | INFO     | ui_testing.pages.order_page:get_suborder_data:409 - getting main order data
2020-09-03 13:28:16.329 | INFO     | ui_testing.pages.order_page:get_suborder_data:416 - getting suborders data
2020-09-03 13:28:17.494 | INFO     | ui_testing.pages.order_page:create_new_suborder:358 - Add new suborder.
2020-09-03 13:28:18.007 | INFO     | ui_testing.pages.order_page:create_new_suborder:364 - Set material type : Raw Material
2020-09-03 13:28:25.707 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:27.108 | INFO     | ui_testing.pages.order_page:create_new_suborder:367 - Set article name : f7a7c378b9
2020-09-03 13:28:27.112 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:37.853 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:38.844 | INFO     | ui_testing.pages.order_page:create_new_suborder:375 - Set test plan : 018117fb3a
2020-09-03 13:28:46.530 | INFO     | ui_testing.pages.order_page:create_new_suborder:375 - Set test plan : dfd1ad35ec
2020-09-03 13:28:54.042 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:54.994 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:56.051 | INFO     | ui_testing.pages.order_page:get_suborder_data:409 - getting main order data
2020-09-03 13:28:56.196 | INFO     | ui_testing.pages.order_page:get_suborder_data:416 - getting suborders data
2020-09-03 13:28:57.820 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-03 13:28:57.823 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:28:58.998 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:01.204 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=84&deleted=0
2020-09-03 13:29:01.634 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-03 13:29:01.642 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_select_large_number_of_test_units_in_one_testplan:2944 - assert the 3 analysis numbers are triggered for each suborder 
2020-09-03 13:29:03.725 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-03 13:29:04.404 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:05.329 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:06.487 | INFO     | ui_testing.pages.orders_page:filter_by_order_no:138 - Filter by order no. : 151-2020
2020-09-03 13:29:07.596 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-03 13:29:07.912 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:08.837 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:09.934 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-03 13:29:11.105 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-03 13:29:20.360 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-03 13:29:23.252 | INFO     | ui_testing.pages.analyses_page:filter_by_order_no:22 - Filter by order number : 151-2020
2020-09-03 13:29:23.257 | INFO     | ui_testing.pages.base_pages:open_filter_menu:90 - open Filter
2020-09-03 13:29:32.719 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-03 13:29:32.756 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2
2020-09-03 13:29:33.991 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
1
2020-09-03 13:29:39.627 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-03 13:29:45.204 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-03 13:29:46.619 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 254.023s

OK

```

### CODE REVIEW

File: api_testing/apis/test_plan_api.py [hunk 1]
```diff
@@ -345,3 +345,31 @@ def set_configuration(self):
         with open(config_file, "r") as read_file:
             payload = json.load(read_file)
         super().set_configuration(payload=payload)
+
+    def create_completed_testplan_with_formatted_article(self, formatted_article, material_type):
```
[the remark this answers is not in the document]
Comment (omnia353) [reply api_testing/apis/test_plan_api.py:349]: i want to specify article and material type i will do that with flags in the original function

[the remark this answers is not in the document]
Comment (omnia353) [reply api_testing/apis/test_plan_api.py]: done

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -2905,4 +2905,65 @@ def test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived(sel
...
+        tst_plan3, tst_plan4, article2 = TestPlanAPI().create_multiple_test_plan_with_same_article()
+        tst_plan5, tst_plan6, article3 = TestPlanAPI().create_multiple_test_plan_with_same_article()
+        test_units1 = [tst_plan2['testUnits'][0]['name'], tst_plan1['testUnits'][0]['name']]
+        test_units2 = [tst_plan3['testUnits'][0]['name'], tst_plan4['testUnits'][0]['name']]
+        test_units3 = [tst_plan5['testUnits'][0]['name'], tst_plan6['testUnits'][0]['name']]
+        self.info(' Create new order.')
+        self.order_page.create_new_order(article=article1, material_type='Raw Material',
+                                         test_plans=[tst_plan1['testPlan']['text'], tst_plan2['testPlan']['text']],
+                                         set_tstunit=False, save=False)
+        order_no = self.order_page.get_no()
```
[the remark this answers is not in the document]
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

**Review by ghada-37 — APPROVED** [review]

**Comment by ghada-37** [comment]
@0xIslamTaha  I reviewed code and that it  matches test case ticket on Jira, I just couldn't run it locally 
when I get this branch locally I found that last test case in orders file in 
`test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived`
So, you'll need to run it locally before merge

**Comment by 0xIslamTaha** [comment]
duplicate

**Pull request closed without merging**